cmake/CompileOptions.cmake

```diff
@@ -105,6 +105,7 @@ if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU" OR "${CMAKE_CXX_COMPILER_ID}" MATCH
         -Wswitch-default
         -Wuninitialized
         -Wmissing-field-initializers
+        -msse4.1
```

Comment:
Seems like this line belongs near ```source/particles/CMakeLists.txt:132```. Please move this compiler flag to this file.
The according code should be like this:
```cmake
$<$<CXX_COMPILER_ID:AppleClang>:
    -msse4.1
    $<$<BOOL:${OPTION_USE_AVX2}>:-mavx2 -mfma>
>
```